mlflow/pyfunc/__init__.py

```diff
@@ -1,19 +1,32 @@
 # -*- coding: utf-8 -*-
 
 """
-The ``mlflow.pyfunc`` module defines a generic filesystem format for Python models and provides
-utilities for saving to and loading from this format. The format is self contained in the sense
-that it includes all necessary information for anyone to load it and use it. Dependencies
-are either stored directly with the model or referenced via a Conda environment.
-
-The convention for pyfunc models is to have a ``predict`` method or function with the following
-signature::
-
-    predict(data: pandas.DataFrame) -> numpy.ndarray | pandas.Series | pandas.DataFrame
-
-This convention is relied on by other MLflow components.
-
-Pyfunc model format is defined as a directory structure containing all required data, code, and
+The ``mlflow.pyfunc`` module defines a generic :ref:`filesystem format <pyfunc-filesystem-format>`
+for Python models and provides utilities for saving to and loading from this format. The format is
```

Comment:
That's really good documentation but I am not sure if it's not too much for in-code docs. I wonder if large bulk of it belongs to the docs instead? 

In my opinion, in-code doc should give functional description necessary to work with the code. E.g. the paragraph starting ```Many of MLflow's model persistence modules ,...``` sounds to me like something that might belong to docs instead of in code. What do you think?

Reply:
This documentation gets rendered and displayed in the "Python API" section of the MLflow docs website. It also provides a reference for how to use the `save_model()` and `log_model()` APIs, so it seems relevant to the code in that sense.

I agree that the `Many of MLflow's model persistence...` portion is probably a better fit for the "Models" documentation section. I've moved it there instead.